net/Socket.hpp

```diff
@@ -470,7 +470,7 @@ class Socket
         if constexpr (!Util::isMobileApp())
         {
 #if ENABLE_DEBUG
-            if (std::getenv("COOL_ZERO_BUFFER_SIZE"))
+            if (std::getenv("COOL_ZERO_BUFFER_SIZE") && _fd >= 0)
```

Comment:
Hmm, but then if isClosed() called twice, you'll get a positive & invalid _fd. Still, this looks OK, I complained about the broader issue at https://github.com/CollaboraOnline/online/pull/10874#issuecomment-2592410559, hopefully soon @Ashod will have time to take a look.